10 спринт.

Kris-found opened the pull request:
Реализовано отображение списка треков в активити search: Реализован Adapter & ViewHolder

Kris-found committed:
10 спринт. Реализовано отображение списка треков в активити search:
Реализован Adapter & ViewHolder

sukhoikms27 commented on app/src/main/res/values-night/themes.xml:
```diff
@@ -18,17 +18,18 @@
-        <item name="android:layout_height">48dp</item>
-        <item name="android:layout_weight">1</item>
-        <item name="android:layout_marginStart">4dp</item>
+        <item name="android:layout_width">match_parent</item>
```
Рекомендация:
В стили нежелательно заносить параметры layout'a, такие как width/height/weight/layout_gravity/layout_margin,: 1) это параметры относящиеся к конкретному экрану, из-за которых будет сложно переиспользовать такие стили на других экранах 
2) из-за того, что эти параметры скрыты в стилях - будет тяжелее дополнять/улучшать экран с новыми элементами

sukhoikms27 commented on app/src/main/java/com/practicum/playlistmaker/TrackViewHolder.kt:
```diff
@@ -0,0 +1,42 @@
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.resource.bitmap.RoundedCorners
+
+fun dpToPx(dp: Float, context: Context): Int {
```
Рекомендация:
Под утилитные методы лучше создать отдельный файл `Utils` например, и держать их там

sukhoikms27 reviewed (commented):
баг с кнопкой очистки поля поиска:
![image](https://github.com/user-attachments/assets/cdaac390-7ed2-4a69-b97c-78c1b6b79780)

Kris-found committed:
10 спринт: исправления
- Добавлен файл Utils для утилитных методов/классов
- баг: поправлено расположение кнопки очистки ввода поиска
